### PR TITLE
wgengine/magicsock: add controlknob tunable for session timeout experiments

### PR DESCRIPTION
**Pull request by bradfitz**
Updates #TODO

### CODE REVIEW

File: wgengine/magicsock/magicsock.go [hunk 1]
```diff
@@ -2729,6 +2729,15 @@ func portableTrySetSocketBuffer(pconn nettype.PacketConn, logf logger.Logf) {
 	}
 }
 
+func (c *Conn) sessionActiveTimeout() time.Duration {
```
[the remark this answers is not in the document]
Comment (bradfitz) [reply wgengine/magicsock/magicsock.go:2732]: It's not a global. That changed a month ago in https://github.com/tailscale/tailscale/pull/9352

With multiple tsnet.Servers in one process you can have multiple magicsock.Conns in the same process with different controlknobs properties.

**Comment by valscale** [comment]
Hi Brad, before you merge this to main, I want to discuss the proper allocation of credit for this work. The work that went into understanding that the session timeout was causing problems was done by other people, and this specific feature is already in our task planning system and assigned to my team. Let's discuss this with our respective management chains. Thank you!

**Comment by bradfitz** [comment]
This was from a conversation I was having with @knyar. The "Updates #TODO" was because I didn't have the full context.

But I also don't think this is the right fix. It was just an exploration of plumbing control knobs with the new CapMap RawMessage arguments which we didn't have prior.

**Comment by valscale** [comment]
As you are aware, this project was intended as a starter project for a new member of my team. Let's discuss this with our respective management teams before you do any additional work on this project.

**Pull request closed without merging**